src/network/rpc/RpcServer.h

```diff
@@ -16,14 +17,19 @@ class RpcHandler;
 
 class RpcServer : public std::enable_shared_from_this<RpcServer>, public jsonrpc::AbstractServerConnector {
  public:
-  RpcServer(boost::asio::io_context &io, boost::asio::ip::tcp::endpoint ep, addr_t node_addr);
+  enum class ServerType { RpcType, GraphQlType };
+
```

Comment:
one more idea, which I leave to you how you decide :) RpcServer could even become base class and then JsonRpcServer and GraphQlServer might be derived from that and differenciate(override virtual func) "read() ".

Reply:
@JakubFornadel I would think graphql over websocket will be needed for the explorer to function efficiently at high throughput

Reply:
Ok, I will add the WS server for graphql as well